src/contracts/atlas/GasAccounting.sol

```diff
@@ -237,48 +252,58 @@ abstract contract GasAccounting is SafetyLocks {
             gasUsed += _getCalldataCost(solverOp.data.length);
         }
 
-        gasUsed = gasUsed * (SURCHARGE_SCALE + ATLAS_SURCHARGE_RATE + BUNDLER_SURCHARGE_RATE) / SURCHARGE_SCALE;
-
         // Calculate what the solver owes
         // NOTE: This will cause an error if you are simulating with a gasPrice of 0
         if (!result.updateEscrow()) {
             // CASE: Solver is not responsible for the failure of their operation, so we blame the bundler
             // and reduce the total amount refunded to the bundler
-            writeoffs += gasUsed;
+            writeoffs += gasUsed.withAtlasAndBundlerSurcharges();
         } else {
             // CASE: Solver failed, so we calculate what they owe.
-            uint256 deficit = _assign(solverOp.from, gasUsed, false);
-            if (deficit > 0) {
-                // Write off any deficit as a gas loss to the bundler so that other solvers aren't forced to pay it.
-                writeoffs += deficit;
-            }
+            _assign(solverOp.from, gasUsed.withAtlasAndBundlerSurcharges(), false);
         }
     }
 
     /// @param ctx Context struct containing relavent context information for the Atlas auction.
     /// @param solverGasLimit The maximum gas limit for a solver, as set in the DAppConfig
-    function _getAdjustedClaimsAndWriteoffs(
+    function _adjustAccountingForFees(
         Context memory ctx,
         uint256 solverGasLimit
     )
         internal
-        view
-        returns (uint256 _claims, uint256 _writeoffs)
+        returns (
+            uint256 _withdrawals,
+            uint256 _deposits,
+            uint256 _claims,
+            uint256 _writeoffs,
+            uint256 netAtlasGasSurcharge
+        )
     {
+        uint256 _surcharge = cumulativeSurcharge;
+
         _claims = claims;
         _writeoffs = writeoffs;
+        _withdrawals = withdrawals;
+        _deposits = deposits;
 
         uint256 _gasLeft = gasleft(); // Hold this constant for the calculations
 
-        uint256 _gasRemainder =
-            _gasLeft * tx.gasprice * (SURCHARGE_SCALE + ATLAS_SURCHARGE_RATE + BUNDLER_SURCHARGE_RATE) / SURCHARGE_SCALE;
-        _claims -= _gasRemainder;
+        // Estimate the unspent, remaining gas that the Solver will not be liable for.
+        uint256 _gasRemainder = _gasLeft * tx.gasprice; // * (SURCHARGE_SCALE + ATLAS_SURCHARGE_RATE +
+            // BUNDLER_SURCHARGE_RATE) / SURCHARGE_SCALE;
```

Comment:
Should we remove the comment?

Reply:
Yes, think now that we've moved to AccountingMath lib for all the fee calcs, the comment can go

Reply:
https://github.com/FastLane-Labs/atlas/pull/281/commits/61bcac3d184d807e1541aa98f98bf21aeb3444eb